deploy/deploy.go

```diff
@@ -280,6 +281,15 @@ func (d *Deployment) Deploy(ctx context.Context, kubecfg string) (rerr error) {
 			rerr = w.Cleanup(rerr)
 		}()
 	}
+	if w, err := events.NewWatcher(ctx, kClient, cancel); err != nil {
```

Comment:
Adding a comment on why event watching is needed here would help I guess.

Reply:
done